unitTests/ui/PagerTest.ts

```diff
@@ -38,6 +38,33 @@ export function PagerTest() {
       expect(test.env.queryController.executeQuery).toHaveBeenCalledTimes(4);
     });
 
+    it('should not be possible to set current page to an invalid value', () => {
+      test.cmp.setPage('a' as any);
+      expect(test.cmp.currentPage).toBe(1);
+      test.cmp.setPage('1' as any);
+      expect(test.cmp.currentPage).toBe(1);
+      test.cmp.setPage(2 as any);
+      expect(test.cmp.currentPage).toBe(2);
+      test.cmp.setPage(1.7 as any);
+      expect(test.cmp.currentPage).toBe(2);
+      test.cmp.setPage(1.5 as any);
+      expect(test.cmp.currentPage).toBe(2);
+      test.cmp.setPage(1.499999 as any);
```

Comment:
In the tests we are using `setPage`, but in the source, `currentPage` method is also public and available. Moreover, `setPage` also contains additional sanitization logic that is not in `currentPage`. If the tests were using `currentPage`, some of these cases would fail.

We should either move everything to the new `currentPage` setter, or keep `currentPage` a property and tell people they need to use the `setPage` method to update the value of `currentPage`.